src/Corax/Querying/IndexSearcher.cs

```diff
@@ -120,18 +120,18 @@ private IndexSearcher(IndexFieldsMapping fieldsMapping)
             _fieldMapping = fieldsMapping;
         }
     }
-    
-    public EntryTermsReader GetEntryTermsReader(long id, ref Page p)
+
+    public void GetEntryTermsReader(long id, ref Page p, out EntryTermsReader reader, CompactKey existingKey = null)
```

Comment:
Can you  please elaborate more on this change? What is the motivation for returning the result via `out` instead of `return`?

Reply:
Copying `EntryTermsReader` is costly (because it is big). By forcing the registration in the stack of the caller function, we avoid the copy when the call returns.

Reply:
Oh, so this is related directly to the fact that `EntryTermsReader ` is a struct.

Reply:
Taking into account that we're already operating on `EntryTermReader` in a reusable manner, perhaps we should transform it into a class and use a single instance of it. We have a `Reset()` method that resets the current state, so we could add a `ChangeDocument(addr, size, dictId)` method to change the pointer and call `Reset` inside. This way, we wouldn't have to worry about copying or disposing.

Reply:
I lean against that on semantic grounds. The process of resetting the object introduces problems around ensuring correct instantiation, resets, etc. By pairing the creation of the object and also forcing to become readonly (as possible) we are gaining on several grounds. On the reliability, we are going to be creating the object directly from the call and we are not going to be able to keep old state around. On the performance, there is no upside of using a heap object vs a stack object. Also the reset requires us to refresh almost everything anyways, we are not saving operations either. The more we can keep in the stack and avoid false sharing between calls/processes the better.